### ISSUE TITLE
vscode: allow specifying path to executable in extension settings

### ISSUE THREAD

**Issue by gilmeir** [oxc-project/oxc#12849]
## Motivation
* Avoid unexpected lint errors suddenly starting to show in vscode after new versions of the extension are released, when auto update is enabled, and oxlint is used as a dev dependency, due to an inconsistency between the extension version and the dev dependency version
* I prefer not to give it ongoing attention by either
  - Telling my team to update the Oxc extension every time I update the version in my project.
  - Turning off those new rules every time

## Example
Since the extension version 1.9.0 was released I suddenly started seeing errors in vscode (on existing code), on a violation of
```
typescript-eslint(explicit-module-boundary-types): Missing argument type on function
```

This rule started producing errors by default since 1.9.0 for those using the "typescript" plugin.

In my project oxlint is installed as a dev dependency locked on 1.8.0, and it passed without errors. I'm running it this way:
```bash
pnpm oxlint --deny-warnings --report-unused-disable-directives
```

I'm not using pnpx as suggested in the docs because I like the visibility of all my deps and their versions being in one place in package.json dependencies/devDependencies.

## Suggestion
1. Add a new settings property to allow users configure how vscode runs oxlint, something like
```json
  "oxc.executablePath": "./node_modules/.bin/oxlint",
```
2. Define priorities, for example:
   1. Project dependency
   2. Executable path in settings
   3. Global dependency
   4. Remote-latest

**Comment by Sysix** [comment]
You can use `oxc.path.server` setting: https://github.com/oxc-project/oxc/blob/main/editors/vscode/README.md
The string can be: `./nodes_modules/.bin/oxc_language_server` or with an `.exe` suffix for Windows file.

**Comment by ahutchings** [comment]
@Sysix When I set `oxc.path.server` to a relative path - `./node_modules/.bin/oxc_language_server` or `node_modules/.bin/oxc_language_server` - I get this error in the extension output:

```
2025-08-28 11:12:19.442 [error] Invalid bin path: ./node_modules/.bin/oxc_language_server A system error occurred (ENOENT: no such file or directory, access './node_modules/.bin/oxc_language_server')
```

An absolute path works fine though: `/Users/<me>/Projects/<project>/node_modules/.bin/oxc_language_server`

Would it be possible to support a relative path for this setting, so that my team can share the `oxc.path.server` config? We check in the `.vscode/settings.json` file in our project.

**Comment by donaldpipowitch** [comment]
I see https://github.com/oxc-project/oxc/pull/13542 was merged an hour ago. This only covers the case with the relative path, correct? I could not share `oxc.path.server` as a project wide setting in my `.vscode/settings.json`, if I have a mix of Windows and non-Windows developers as I sometimes would have to use `.exe` and sometimes not. (Referring to [this comment](https://github.com/oxc-project/oxc/issues/12849#issuecomment-3160832991).)

**Comment by Sysix** [comment]
Thanks for the notice!
Yes, this could and should be probably optimized by the extension 👍 
Will look into it :)

**Comment by donaldpipowitch** [comment]
Thank you 😊

**Comment by Sysix** [comment]
The work is a big bigger than expected, created a new issue for it: https://github.com/oxc-project/oxc/issues/14167
Closing this one.
